### DISCUSSION TITLE
Consider making `RavEngine::World`'s destructor virtual

### DISCUSSION THREAD

**Discussion by LouChiSoft** [RavEngine/RavEngine#13] [Ideas]
The current non-virtual destructor for world means that only the base class destructor gets called when a world is destroyed. This means that unless I missed a function somewhere there is no way for a user to define a derived world that cleans up manually allocated resources.
Changing `RavEngine::World`'s destructor to virtual would allow users to create custom destructors for their derived classes that are called before calling the base class destructor.

**Comment by Ravbug** [comment]
Good catch. You're welcome to make a pull request if you would like.

**Reply by LouChiSoft** [reply]
Sure, let me just figure out how. Never done a GitHub PR before

**Reply by Ravbug** [reply]
Maybe this will help? https://www.ravbug.com/tutorials/github-pr/

**Reply by LouChiSoft** [reply]
Thanks. 👍🏻
